### ISSUE TITLE
Confusion Matrix

### ISSUE THREAD

**Issue by TillJohanndeiter** [keras-team/keras#15187]
**System information**.

TensorFlow version (you are using): 2.5.0
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

Print confusion matrix for muticlass prediction. Matrix represents prediction to ground truth

Multiclass predictions cannot be comfortable evaluated with respect to associated classes with current metrics (Acc, Precison, Recall...). They do not show if the prediction is balanced. Especially comparatively small classes are not represented. Currently you have to code a confusion matrix by yourself or use another library to gain these information.

**Will this change the current api? How?**

Callback _PrintConfusionMatrix_ is in _keras.callbacks_ 

**Who will benefit from this feature?**

* Evaluators/Trainers of a multi class predictions to find weak predicted classes

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes 
- Briefly describe your candidate solution(if contributing): Implement callback to print matrix for passed dataset after training. 
Optional _id_to_class dict_ can be passed to print actual labels.

**Comment by mattdangerw** [comment]
This is not something we would want to carry in core Keras right now, but https://github.com/tensorflow/addons may be a good place for this. You could open an issue there and see what feedback you get. Thank you!
